### PR TITLE
MULE-16124: Deadlock in JMS connector when using bean defined CachingConnectionFactory and WMQ Broker

### PR DESCRIPTION
**Pull request by plbalbi**

### CODE REVIEW

**Comment by estebanwasinger** [comment]
this issue is also impacting Mule 4?

File: transports/jms/src/main/java/org/mule/transport/jms/CannotReceiveMessageWhenHandlingExceptionException.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.transport.jms;
+
+import org.mule.api.MuleException;
```
Comment (sonar-mulesoft) [inline transports/jms/src/main/java/org/mule/transport/jms/CannotReceiveMessageWhenHandlingExceptionException.java]: Code Smell Code Smell: Remove this unused import 'org.mule.api.MuleException'. ([squid:UselessImportCheck](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AUselessImportCheck))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7461&issues=AWhqv6O7otIXnmsUmEJU&open=AWhqv6O7otIXnmsUmEJU)

[ISSUE_KEY]: AWhqv6O7otIXnmsUmEJU

Comment (sonar-mulesoft) [inline transports/jms/src/main/java/org/mule/transport/jms/CannotReceiveMessageWhenHandlingExceptionException.java]: Code Smell Code Smell: Remove this unused import 'org.mule.api.MuleException'. ([squid:UselessImportCheck](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AUselessImportCheck))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7461&issues=AWhqv6O7otIXnmsUmEJU&open=AWhqv6O7otIXnmsUmEJU)

[ISSUE_KEY]: AWhqv6O7otIXnmsUmEJU

Comment (sonar-mulesoft) [inline transports/jms/src/main/java/org/mule/transport/jms/CannotReceiveMessageWhenHandlingExceptionException.java]: Code Smell Code Smell: Remove this unused import 'org.mule.api.MuleException'. ([squid:UselessImportCheck](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AUselessImportCheck))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7461&issues=AWhqv6O7otIXnmsUmEJU&open=AWhqv6O7otIXnmsUmEJU)

[ISSUE_KEY]: AWhqv6O7otIXnmsUmEJU

Comment (sonar-mulesoft) [inline transports/jms/src/main/java/org/mule/transport/jms/CannotReceiveMessageWhenHandlingExceptionException.java]: Code Smell Code Smell: Remove this unused import 'org.mule.api.MuleException'. ([squid:UselessImportCheck](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AUselessImportCheck))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7461&issues=AWhqv6O7otIXnmsUmEJU&open=AWhqv6O7otIXnmsUmEJU)

[ISSUE_KEY]: AWhqv6O7otIXnmsUmEJU

**Pull request closed without merging**